src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,138 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    public User createUser(String login, String password, int age) {
+        User user = new User();
+        user.setLogin(login);
+        user.setPassword(password);
+        user.setAge(age);
+        return user;
+    }
+
+    @Test
+    void loginExist_NotOk() {
+        User user1 = createUser("gooodlogin@gmail.com", "bayraktar",
+                27);
+        registrationService.register(user1);
+        User user2 = createUser("gooodlogin@gmail.com", "123456", 64);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user2);
+        });
+    }
+
+    @Test
+    void checkAge_Ok() {
+        User user1 = createUser("gooooodlogin@gmail.com", "bayraktar",
+                27);
+        registrationService.register(user1);
+        assertEquals(storageDao.get("gooooodlogin@gmail.com"), user1);
+    }
+
+    @Test
+    void checkAge_NotOk() {

```

Comment:
change method names according to convention
```suggestion
    void register_underage_notOk() {

```